### ISSUE TITLE
Spawned process isn't killed when running via package.json scripts.

### ISSUE THREAD

**Issue by RossMcMillan92** [oven-sh/bun#7441]
### What version of Bun is running?

1.0.15+b3bdf22eb

### What platform is your computer?

Linux 5.15.133.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

Repo: https://github.com/RossMcMillan92/bun-spawn-bug-2

``` bash
bun install
```

``` bash
# Note that this works as expected.
# The first counter ends when the app is killed
# and only the newly spawned counter is logged
bun ./src/watcher.ts
```
<img width="188" alt="image" src="https://github.com/oven-sh/bun/assets/3329319/b0ff3793-cb2e-41e5-b6e9-a6476abf13b9">

``` bash
# Note that this runs the exact same script as
# above, but the first timer doesn't stop counting.
# The only difference is we run ./src/watcher.ts via
# the start script instead of calling it directly.
bun run start
```

<img width="191" alt="image" src="https://github.com/oven-sh/bun/assets/3329319/388a7a0c-3bcc-4f38-9a49-9677c1976724">

### What is the expected behavior?

Spawned processes are killed when `spawnedProcess.kill()` is called.

### What do you see instead?

It seems to work when calling the script directly `bun ./src/watcher.ts`, but not when called via `bun run start` (where start script is `bun ./src/watcher.ts`)

**Comment by bujoralexandru** [comment]
Has this issue been addressed somehow, maybe in a release that dates the one mentioned by the issue author? I encounter a similar behaviour in my project.

**Comment by Jarred-Sumner** [comment]
This was addressed in Bun v1.1.27. We made signals from the root propagate down to the child. 

If you’re still running into this, please provide a reproduction and maybe open a new issue
